Javascript/getMaxSumOfDigits.js

```diff
@@ -0,0 +1,41 @@
+/*
+자릿수의 합
+N개의 자연수가 입력되면 각 자연수의 자릿수의 합을 구하고, 그 합이 최대인 자연수를 출력 
+하는 프로그램을 작성하세요. 자릿수의 합이 같은 경우 원래 숫자가 큰 숫자를 답으로 합니다. 
+만약 235 와 1234가 동시에 답이 될 수 있다면 1234를 답으로 출력해야 합니다.
+- 입력설명
+첫 줄에 자연수의 개수 N(3<=N<=100)이 주어지고, 그 다음 줄에 N개의 자연수가 주어진다. 
+각 자연수의 크기는 10,000,000를 넘지 않는다.
+- 출력설명
+자릿수의 합이 최대인 자연수를 출력한다. 
+- 입력예제 1
+128 460 603 40 521 137 123 
+- 출력예제 1
+137
+*/
+
+const solution = (arr) => {
+    let answer = 0;
+    let max = Number.MIN_SAFE_INTEGER;
+
+
+    for (num of arr) {
+        let sum = 0, temp = num;
+        while (temp) {
+            sum += temp % 10;
+            temp = Math.floor(temp / 10);
+        }
+        if (sum > max) {
+            max = sum;
+            answer = num;
+        }
+        else if (sum === max)
+            if (num > answer) answer = num;
```

Comment:
```suggestion
        if (sum > max) {
            max = sum;
            answer = num;
        } else if (sum === max && num > answer) {
            answer = num;
        }
```
if...else if 문을 정돈해봤어요.

개인적으로 같은 depth의 if...else if면 한쪽에 중괄호를 사용했다면 다른쪽에도 같이 적용하는 것이 훨씬 가독성이 좋을 것 같아요.

Reply:
그러네요 굳이 한 뎁스 더 들어갈 필요가 없어 보이네요! ㅎㅎ 
좋은 의견 감사합니다~

Reply:
한 뎁스 더 들어가는 것도 그렇고 중괄호 생략에 관한 것도 포함한 의견이었어요